include/ssm-internal.h

```diff
@@ -154,16 +154,36 @@ enum ssm_kind {
  *  alloc_mem_handler. These handlers may also assume they will not be invoked
  *  to request memory ranges of less than #SSM_MEM_POOL_MAX bytes.
  *
+ *  If the allocator is compiled with valgrind support (i.e., without defining
+ *  @a NVALGRIND), it will perform a leak-check summary, to checkpoint how much
+ *  memory has already been allocated.
+ *
  *  @platformonly
  *
- *  @param alloc_page_handler   allocates pages.
- *  @param alloc_mem_handler    allocates arbitrarily large.
- *  @param free_mem_handler     frees pages allocated with @a alloc_mem_handler.
+ *  @param alloc_page_handler allocates pages.
+ *  @param alloc_mem_handler  allocates arbitrarily large.
+ *  @param free_mem_handler   frees memory allocated with @a alloc_mem_handler.
  */
 void ssm_mem_init(void *(*alloc_page_handler)(void),
                   void *(*alloc_mem_handler)(size_t),
                   void (*free_mem_handler)(void *, size_t));
 
+/** @brief Tears down the underlying allocator system.
```

Comment:
I added this endpoint to give us the opportunity to deallocate pages in the future.

We might not be able to do this very efficiently, but that's ok. In my view, this exists primarily for testing purposes.